pypeit/edgetrace.py

```diff
@@ -3970,326 +3988,235 @@ def fully_masked_traces(self, flag=None, exclude=None):
-                :attr:`files` is also None, the method will raise an
-                exception.
+            offsets_range (:obj:`list`):
+                range of offsets in pixels allowed between the slit positions predicted by
+                 the mask design and the traced slit positions.
+            step (:obj:`int`):
+                step in pixels used to generate a list of possible offsets within
+                the `offsets_range`:
+
             debug (:obj:`bool`, optional):
                 Run in debug mode.
         """
-        # Still not done with this function...
-        raise NotImplementedError()
 
-        # Check that there are traces to refine!
+        # Check that there are traces to match!
         if self.is_empty:
-            msgs.error('No traces to refine.')
+            msgs.error('No traces to match.')
 
         # The PCA decomposition must have already been determined
         if self.pcatype is None:
             msgs.error('Must first run the PCA analysis for the traces; run build_pca.')
 
-        # Get the file to use when parsing the mask design information
-        _design_file = (None if self.traceimg.files is None else self.traceimg.files[0]) \
-                            if design_file is None else design_file
-        if _design_file is None or not os.path.isfile(_design_file):
-            msgs.error('Slit-mask design file not found or none provided.')
-
-        # Get the paramters to use
-        maxiter = self.par['mask_reg_maxiter']
-        maxsep = self.par['mask_reg_maxsep']
-        sigma = self.par['mask_reg_sigrej']
-        ignore_alignment = self.par['ignore_alignment']
-
-        # TODO: Set allow_resync and design_file to be a parameters, as
-        # well?
-
-        # Read the design data
-        msgs.info('Reading slit-mask design information from: {0}'.format(_design_file))
-        if self.spectrograph.get_slitmask(_design_file) is None:
-            msgs.error('Unable to read design file or no slit-mask design reader '
-                       'defined for {0}.'.format(self.spectrograph.spectrograph))
-
-        # Match both left and right edges simultaneously
-        x_design = np.array([self.spectrograph.slitmask.bottom[:,0],
-                             self.spectrograph.slitmask.top[:,0]]).T.ravel()
-        reference_row = self.left_pca.reference_row if self.par['left_right_pca'] \
-                            else self.pca.reference_row
-        x_det = self.edge_fit[reference_row,:]
-
+        # `traceimg` must have knowledge of the flat frame that built it
+        if self.spectrograph.get_slitmask(self.traceimg.files[0]) is None:
+            msgs.error('Unable to read slitmask design info')
+        if self.spectrograph.get_grating(self.traceimg.files[0]) is None:
+            msgs.error('Unable to read grating info')
+        if self.spectrograph.get_amapbmap(self.traceimg.files[0]) is None:
+            msgs.error('Unable to read amap and bmap')
+
```

Comment:
Avoid using x and y to describe positions in PypeIt imags. The preferred terminology is always, (spec, spat), where spectral is the first dimension of PypeIt spectra, which is always spectral and spat is the second dimension of PypeIt oriented spectra which is always spatial. By sticking to this convention  you avoid ambiguity between flipped images in python, IDL, etc.